src/base64.rs

```diff
@@ -0,0 +1,730 @@
+/// Provide an upper limit on how much space could be required
```

Comment:
I see you just added these files here without integrating them in the crate (via `mod` declarations), which you deferred to the next commit (at least for one of the modules?). This all seems a little iffy in terms of our usual goal of having each commit pass CI... Since IIRC `base64` hasn't made it into the committed `main` history of the pemfile repo, I wouldn't worry too much about this separate step.

So I think I would suggest squashing the first three commits together -- I don't think there are substantial changes here that are worth keeping separate.